File: src/main/java/org/racket/lang/core/RacketFileType.java
package org.racket.lang.core;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class RacketFileType extends LanguageFileType {
    public static final RacketFileType INSTANCE = new RacketFileType();

    private RacketFileType() {
        super(RacketLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Racket file";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Racket language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "rkt";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return RacketIcons.FILE;
    }
}

File: src/main/kotlin/org/racket/lang/core/RacketFileType.kt
package org.racket.lang.core

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

class RacketFileType private constructor() : LanguageFileType(RacketLanguage) {
    override fun getName(): String {
        return "Racket file"
    }

    override fun getDescription(): String {
        return "Racket language file"
    }

    override fun getDefaultExtension(): String {
        return "rkt"
    }

    override fun getIcon(): Icon? {
        return RacketIcons.FILE
    }

    companion object {
        val INSTANCE = RacketFileType()
    }
}
